lib/active_hash/base.rb

```diff
@@ -188,6 +189,7 @@ def create(attributes = {})
 
       def create!(attributes = {})
         record = new(attributes)
+        validate_unique_id(record)
```

Comment:
Does the create! method allows to rewrite an already existing id ?

Reply:
unfortunately, `create!` is broken since it doesn't "modify" or "overwrite" due to `insert` not being called because `save` calls insert only if the record doesn't exists. Bottom line is that `create` and `create!` are the same except one doesn't `mark_dirty`. 

Basically the "bang" logic is broken and I didn't want to clutter this PR but maybe we can  at least add `if dirty` condition to `create!` but this would just hide the issue. At least with the current modification, if someone would call `create!` assuming it would overwrite they would get alerted; otherwise, it fails silently as if everything is fine.

What do you think @braktar `create!` is not used anyways and this way is at least safe? I think this create/save/insert and their "banged" versions needs an overhaul altogether.

Reply:
It is indeed better to have an alert instead of a silent error.
Then, I agree with your proposition